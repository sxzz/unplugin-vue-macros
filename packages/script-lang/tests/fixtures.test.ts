import { resolve } from 'node:path'
import { describe } from 'vitest'
import {
  RollupEsbuildPlugin,
  RollupVue,
  rollupBuild,
  testFixtures,
} from '@vue-macros/test-utils'
import VueScriptLang from '../src/rollup'

describe('fixtures', async () => {
  await testFixtures(
    'tests/fixtures/*.{vue,js,ts}',
    (args, id) =>
      rollupBuild(id, [
        VueScriptLang(),
        RollupVue({
          isProduction: args.isProduction,
          inlineTemplate: args.isProduction,
        }),
        RollupEsbuildPlugin({
          target: 'esnext',
        }),
      ]),
    {
      cwd: resolve(__dirname, '..'),
      promise: true,
      params: [['isProduction', [true, false]]],
    },
  )
})
