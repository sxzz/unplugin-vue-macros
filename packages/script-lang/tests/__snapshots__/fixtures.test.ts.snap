// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fixtures > tests/fixtures/basic.vue > isProduction = false 1`] = `
"import { defineComponent, toDisplayString } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

const bar = 1;
var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "basic",
  props: {
    foo: {}
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return toDisplayString(_ctx.foo);
    };
  }
});

function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return toDisplayString($props.foo);
}
var basic = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render], __FILE__]);

export { bar, basic as default };
"
`;

exports[`fixtures > tests/fixtures/basic.vue > isProduction = true 1`] = `
"import { defineComponent, toDisplayString } from 'vue';

const bar = 1;
var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "basic",
  props: {
    foo: {}
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return toDisplayString(_ctx.foo);
    };
  }
});

export { bar, _sfc_main as default };
"
`;
